manager/apis/app/v1alpha1/plotter_types.go

```diff
@@ -4,21 +4,306 @@
+	WriteTrigger SubFlowTrigger = "write"
+)
+
+// Subflows is a list of data flows which are originated from the same data asset
+// but are triggered differently (e.g., one upon init
+// trigger and one upon workload trigger)
+type SubFlow struct {
+	// Name of the SubFlow
+	// +required
+	Name string `json:"name"`
+
+	// Type of the flow (e.g. read)
+	// +required
+	FlowType DataFlow `json:"flowType"`
+
+	// Triggers
+	// +required
+	Triggers []SubFlowTrigger `json:"triggers"`
+
+	// Steps defines a series of sequential/parallel data flow steps
+	// +required
+	Steps []SequentialSteps `json:"steps" protobuf:"bytes,11,opt,name=steps"`
+}
+
+// Flows is the list of data flows driven from fybrikapplication:
+// Each element in the list holds the flow of the data requested in fybrikapplication.
+type Flow struct {
+	// Name of the flow
+	// +required
+	Name string `json:"name"`
+
+	// Type of the flow (e.g. read)
+	// +required
+	FlowType DataFlow `json:"flowType"`
+
+	// AssetID indicates the data set being used in this data flow
+	// +required
+	AssetID string `json:"assetId"`
+
+	// +required
+	SubFlows []SubFlow `json:"subFlows"`
+}
+
+// ModuleInfo is a copy of M4DModule Custom Resource.  It contains information
+// to instantiate resource of type M4DModule.
+type ModuleInfo struct {
+	// Name of the template
+	// +required
+	Name string `json:"name"`
+
+	// Kind of k8s resource
+	// +required
+	Kind string `json:"kind"`
+
+	// Chart contains the information needed to use helm to install the capability
+	// +required
+	Chart ChartSpec `json:"chart"`
+
+	// Capabilities declares what this module knows how to do and the types of data it knows how to handle
+	// The key to the map is a CapabilityType string
+	// +required
+	Capabilities []ModuleCapability `json:"capabilities"`
+}
+
+// Template contains basic information about the required modules to serve the fybrikapplication
+// e.g., the module helm chart name.
+type Template struct {
+	// Template name
+	Name string `json:"name"`
+	// Modules is a list of dependent modules. e.g., if a plugin module is used
+	// then the service module is used in should appear in the modules list of the
+	// same template
+	// +required
+	Modules []ModuleInfo `json:"modules"`
+}
+
+// SubFlowsStatus holds status information about a subFlow
+type SubFlowsStatus struct {
+	// SubFlow name
+	// +required
+	Name string `json:"name"`
+
+	// ObservedState includes information about this subflow
+	// It includes readiness and error indications, as well as user instructions
+	// +optional
+	ObservedState ObservedState `json:"status,omitempty"`
+}
+
+// AssetStatus includes information to be reported back to the FybrikApplication resource
+// It holds the status and access information for each asset in fybrikapplication
+type AssetStatus struct {
+	// Endpoint
+	// +required
+	Endpoint string `json:"endpoint"`
+
+	// Port -- endpoint should contains it already?
+	// +required
+	Port int32 `json:"port"`
```

Comment:
If necessary, I think it should be in the endpoint

Reply:
I used `EndpointSpec` instead